### PR TITLE
[feature-improvement] Improve Identity Verification Providers feature

### PR DESCRIPTION
**Pull request by JayaShakthi97**
### Purpose

$subject

### Related Issues

- https://github.com/wso2/product-is/issues/20956

### Related PRs

- N/A

### Checklist

- [x] e2e cypress tests locally verified. (for internal contributers)
- [x] Manual test round performed and verified.
- [x] UX/UI review done on the final implementation.
- [ ] Documentation provided. (Add links if there are any)
- [ ] Relevant backend changes deployed and verified
- [ ] [Unit tests](/docs/testing/UNIT_TESTING.md) provided. (Add links if there are any)
- [ ] [Integration tests](https://github.com/wso2/product-is/tree/master/modules/integration) provided. (Add links if there are any)

### Security checks
- [x] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines
- [ ] Ran FindSecurityBugs plugin and verified report
- [x] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets

### CODE REVIEW

File: apps/console/src/public/deployment.config.json [hunk 1]
```diff
@@ -660,10 +660,10 @@
                 }
             },
             "identityVerificationProviders": {
-                "enabled": false,
+                "enabled": true,
                 "scopes": {
                     "create": [
-                        "internal_idvp_create"
+                        "internal_idvp_add"
```
Comment (pavinduLakshan) [inline apps/console/src/public/deployment.config.json:666]: Let's update the framework as well.

File: features/admin.connections.v1/components/authenticator-grid.tsx [hunk 2]
```diff
@@ -254,13 +280,26 @@ export const AuthenticatorGrid: FunctionComponent<AuthenticatorGridPropsInterfac
      *
      * @param id - Authenticator ID.
      */
-    const handleAuthenticatorDelete = (id: string): void => {
+    const handleAuthenticatorDelete = (id: string, connectionType: string): void => {
```
Comment (pavinduLakshan) [inline features/admin.connections.v1/components/authenticator-grid.tsx:283]:
```suggestion
    const handleAuthenticatorDelete = (id: string, connectionType: ConnectionTypes): void => {
```

File: features/admin.connections.v1/components/authenticator-grid.tsx [hunk 3]
```diff
@@ -254,13 +280,26 @@ export const AuthenticatorGrid: FunctionComponent<AuthenticatorGridPropsInterfac
...
+            case ConnectionTypes.IDVP:
+                deleteAction = deleteIdentityVerificationProvider;
+
+                break;
+
+            default:
+                deleteAction = deleteConnection;
+
+                break;
+        }
```
Comment (pavinduLakshan) [inline features/admin.connections.v1/components/authenticator-grid.tsx:299]:
```suggestion
        let deleteAction: (id: string) => Promise<AxiosResponse> = connectionType === ConnectionTypes.IDVP
            ? deleteIdentityVerificationProvider;
            : deleteConnection;
```

File: features/admin.connections.v1/components/authenticator-grid.tsx [hunk 4]
```diff
@@ -346,10 +385,55 @@ export const AuthenticatorGrid: FunctionComponent<AuthenticatorGridPropsInterfac
      */
     const handleGridItemOnClick = (e: SyntheticEvent, authenticator: ConnectionInterface
         | AuthenticatorInterface): void => {
-        handleAuthenticatorEdit(authenticator.id);
+        handleAuthenticatorEdit(authenticator.id, authenticator.type);
         onItemClick && onItemClick(e, authenticator);
     };
 
+    /**
+     * Checks if the delete button should be enabled.
```
Comment (pavinduLakshan) [inline features/admin.connections.v1/components/authenticator-grid.tsx:393]:
```suggestion
     * Checks if the connection delete button should be enabled.
```

File: features/admin.connections.v1/components/authenticator-grid.tsx [hunk 5]
```diff
@@ -346,10 +385,55 @@ export const AuthenticatorGrid: FunctionComponent<AuthenticatorGridPropsInterfac
...
+
+        if (!hasConnectionDeletePermission) {
+            return false;
+        }
+
+        return ConnectionsManagementUtils.isConnectorIdentityProvider(authenticator)
+            || (authenticator as ConnectionInterface).type === AuthenticatorTypes.FEDERATED;
+    };
+
+    const resolveResourceImage = (
```
Comment (pavinduLakshan) [inline features/admin.connections.v1/components/authenticator-grid.tsx:411]: Let's add the doc comments for this method

File: features/admin.connections.v1/components/authenticator-grid.tsx (hunk 3 above)
Comment (JayaShakthi97) [reply features/admin.connections.v1/components/authenticator-grid.tsx:299]: IMO, switch case is more readable than the ternary syntax.
And when there are more connection types listed on the connections page (which is a future plan), it'll be more convenient to extend the impl with switch case.

File: features/admin.connections.v1/components/authenticator-grid.tsx (hunk 2 above)
Comment (JayaShakthi97) [reply features/admin.connections.v1/components/authenticator-grid.tsx]: Fixed via 7fe33104b63800db89ecdeb87666136ca897b765

File: features/admin.connections.v1/components/authenticator-grid.tsx (hunk 5 above)
Comment (JayaShakthi97) [reply features/admin.connections.v1/components/authenticator-grid.tsx:419]: Fixed via d89d804b4490c34fa96959d8550072fd7df05403

File: apps/console/src/public/deployment.config.json (hunk 1 above)
Comment (JayaShakthi97) [reply apps/console/src/public/deployment.config.json:666]: Correct scope is already defined in the framework. Issue is only in identity-apps

File: features/admin.connections.v1/components/authenticator-grid.tsx [hunk 6]
```diff
@@ -195,7 +212,16 @@ export const AuthenticatorGrid: FunctionComponent<AuthenticatorGridPropsInterfac
      *
      * @param idpId - Identity provider id.
      */
-    const handleAuthenticatorDeleteInitiation = async (idpId: string): Promise<void> => {
+    const handleAuthenticatorDeleteInitiation = async (idpId: string, connectionType: string): Promise<void> => {
+        // If the connection is an Identity Verification Provider, then skip checking for connected apps.
+        if (connectionType === ConnectionTypes.IDVP) {
+            setDeletingIDP(authenticators.find(
+                (idp: ConnectionInterface | AuthenticatorInterface) => idp.id === idpId)
```
Comment (dasuni-30) [inline features/admin.connections.v1/components/authenticator-grid.tsx:219]:
```suggestion
                (idp: ConnectionInterface | AuthenticatorInterface) => idp?.id === idpId)
```

File: features/admin.connections.v1/components/authenticator-grid.tsx [hunk 7]
```diff
@@ -346,10 +385,63 @@ export const AuthenticatorGrid: FunctionComponent<AuthenticatorGridPropsInterfac
      */
     const handleGridItemOnClick = (e: SyntheticEvent, authenticator: ConnectionInterface
         | AuthenticatorInterface): void => {
-        handleAuthenticatorEdit(authenticator.id);
+        handleAuthenticatorEdit(authenticator.id, authenticator.type);
```
Comment (dasuni-30) [inline features/admin.connections.v1/components/authenticator-grid.tsx:388]:
```suggestion
        handleAuthenticatorEdit(authenticator?.id, authenticator?.type);
```

File: features/admin.connections.v1/components/authenticator-grid.tsx [hunk 8]
```diff
@@ -346,10 +385,63 @@ export const AuthenticatorGrid: FunctionComponent<AuthenticatorGridPropsInterfac
...
+     * @param isIdP - Whether the connection is an IDP.
+     * @param isOrganizationSSOIDP - Whether the connection is an organization SSO IDP.
+     * @returns - Resolved image element.
+     */
+    const resolveResourceImage = (
+        connection: ConnectionInterface,
+        isIdP: boolean,
+        isOrganizationSSOIDP: boolean
+    ): string => {
+        if (connection.type === ConnectionTypes.IDVP) {
```
Comment (dasuni-30) [inline features/admin.connections.v1/components/authenticator-grid.tsx:424]:
```suggestion
        if (connection?.type === ConnectionTypes.IDVP) {
```

File: features/admin.identity-verification-providers.v1/api/identity-verification-provider.ts [hunk 9]
```diff
@@ -129,69 +142,40 @@ export const createIdentityVerificationProvider = (data: IdentityVerificationPro
...
+export const getIdentityVerificationProvidersList = (
+    limit: number,
+    offset: number
+): Promise<IdVPListResponseInterface> => {
     const requestConfig: RequestConfigInterface = {
         headers: {
-            "Accept": AcceptHeaderValues.APP_JSON,
-            "Content-Type": ContentTypeHeaderValues.APP_JSON
+            "Accept": "application/json",
+            "Access-Control-Allow-Origin": store.getState().config.deployment.clientHost,
```
Comment (dasuni-30) [inline features/admin.identity-verification-providers.v1/api/identity-verification-provider.ts:159]:
```suggestion
            "Access-Control-Allow-Origin": store.getState().config?.deployment?.clientHost,
```

File: features/admin.identity-verification-providers.v1/api/use-get-idvp-template-list.ts [hunk 10]
```diff
@@ -0,0 +1,57 @@
...
+import { ConnectionTemplateInterface } from "@wso2is/admin.connections.v1/models/connection";
+import useRequest, {
+    RequestConfigInterface,
+    RequestErrorInterface,
+    RequestResultInterface
+} from "@wso2is/admin.core.v1/hooks/use-request";
+import { store } from "@wso2is/admin.core.v1/store";
+import { AcceptHeaderValues, HttpMethods } from "@wso2is/core/models";
+
+export const useGetIdVPTemplateList = <
```
Comment (dasuni-30) [inline features/admin.identity-verification-providers.v1/api/use-get-idvp-template-list.ts:28]: Let's add a comment

File: features/admin.identity-verification-providers.v1/api/use-get-idvp-template.ts [hunk 11]
```diff
@@ -0,0 +1,50 @@
...
+import useRequest, {
+    RequestConfigInterface,
+    RequestErrorInterface,
+    RequestResultInterface
+} from "@wso2is/admin.core.v1/hooks/use-request";
+import { store } from "@wso2is/admin.core.v1/store";
+import { AcceptHeaderValues, HttpMethods } from "@wso2is/core/models";
+import { IdVPTemplateInterface } from "../models/identity-verification-providers";
+
+export const useGetIdVPTemplate = <Data = IdVPTemplateInterface, Error = RequestErrorInterface>(
```
Comment (dasuni-30) [inline features/admin.identity-verification-providers.v1/api/use-get-idvp-template.ts:28]: Let's add a comment

**Review by dasuni-30 — APPROVED** [review]

**Comment by JayaShakthi97** [comment]
Rest of the review comments will be addressed via a followup PR.

**Pull request merged**